### DISCUSSION TITLE
Best way for an asset_sensor to trigger a job containing the materialized partition data?

### DISCUSSION THREAD

**Discussion by EtienneT** [dagster-io/dagster#18257] [Q&A]
Here is a scenario, once a daily partitioned asset gets materialized, I need to call an API endpoint to submit the data for the day.

How can I do that using an @asset_sensor and a job?  The job would need to have access to the latest materialized partition from the asset.

**Comment by clairelin135** [comment]
Hi @EtienneT. I'd recommend passing the partition as config to your job, something like this:
```python
class MyConfig(Config):
    partition: str

@op
def my_op(config: MyConfig):
    pass

@job
def my_job():
    my_op()


@asset_sensor(asset_key=my_asset.key, job=my_job)
def my_asset_sensor(context: SensorEvaluationContext, asset_event: EventLogEntry):
    yield RunRequest(run_config={"ops": {"my_op": {"config": {"partition": asset_event.asset_materialization.partition}}}})
```

**Reply by EtienneT** [reply]
Can my_op have a dependency to a partionned asset though?  Because I would need to have access to the data corresponding to the partiton of an asset in my_op.

**Reply by clairelin135** [reply]
@EtienneT here's an example: https://docs.dagster.io/concepts/ops-jobs-graphs/graphs#loading-an-asset-as-an-input

I think by default, if the job is unpartitioned, it loads all partitions of the asset
